meta/store.go

```diff
@@ -129,6 +129,12 @@ func (s *Store) IDPath() string { return filepath.Join(s.path, "id") }
 
 // Open opens and initializes the raft store.
 func (s *Store) Open() error {
+	// Verify that no more than 3 peers.
+	// https://github.com/influxdb/influxdb/issues/2750
+	if len(s.peers) > 3 {
```

Comment:
Peers are hardcoded in the configuration file. All nodes need to have the same set of peers so there's not really a sense of "joining" anymore. If you start the first server up and you list 4 peers then it'll immediately fail.